except_star.md

```diff
@@ -1,31 +1,172 @@
+
+## Motivation
+
+The interpreter is currently able to propagate at most one exception at a time.
+The chaining features introduced in PEP3134 [reference] link together exceptions
+that are related to each other as the cause or context, but there are situations
+where there are multiple unrelated exceptions that need to be propagated together
+as the stack unwinds. Several real world use cases are listed below.
+
+[TODO: flesh these out]
+* asyncio programs, trio, etc
+
+* Multiple errors from separate retries of an operation [https://bugs.python.org/issue29980]
+
+* Situations where multiple unrelated exceptiion may be of interest to calling code [https://bugs.python.org/issue40857]
+
+* Multiple teardowsn in pytest raising exceptions [https://github.com/pytest-dev/pytest/issues/8217]
+
+## Rationale
+
+Grouping several exceptions together can be done without changes to the language, simply by creating
+a container exception type. Trio is an example of a library that has made use of this technique in
+its `MultiError` type [reference to Trio MultiError]. However, such approaches require calling code
+to catch the container exception type, and then inspect it to determine the types of errors that had
+occurred, extract the ones it wants to handle and reraise the rest.
+
+Changes to the language are required in order to extend support for `ExceptionGroup`s in the style
+of existing exception handling mechanisms. At the very least we would like to be able to catch an
+`ExceptionGroup` only if it contains an exception type that we that chose to handle. Exceptions of
+other types in the same `ExceptionGroup` need to be automatically reraised, otherwise it is too easy
+for user code to inadvertently swallow exceptions that it is not handling.
+
+The purpose of this PEP, then, is to add the `except*` syntax for handling `ExceptionGroups`s in
+the interpreter, which in turn requires that `ExceptionGroup` is added as a builtin type. The
+semantics of handling `ExceptionGroup`s are not backwards compatible with the current exception
+handling semantics, so could not modify the behaviour of the `except` keyword and instead added
+the new `except*` syntax.
+
+
+## Specification
+
+### ExceptionGroup
+
+The new builtin exception type, `ExceptionGroup` is a subclass of `BaseException`,
+so it is assignable to `Exception.__cause__` and `Exception.__context__`, and can
+be raised and handled as any exception with `raise ExceptionGroup(...)` and
+`try: ... except ExceptionGroup: ...`.
+
+Its constructor takes two parameters: a message string and a sequence of the nested
+exceptions, for example:
+`ExceptionGroup('many problems', [ValueError('bad value'), TypeError('bad type')])`.
+
+The ExceptionGroup class exposes these parameters in the the fields `msg` and
+`excs` (TODO: did we want to rename excs?).  A nested exception can also be an
+`ExceptionGroup` so the class represents a tree of exceptions (to avoid
+cyclic graph we make the `excs` field immutable?).
```

Comment:
Not sure about making it immutable. I love immutability, but we let users change virtually all other attributes on exceptions. It would be weird to single out just the `errors` field.